src/adapter/src/catalog.rs

```diff
@@ -506,8 +506,15 @@ impl CatalogState {
     /// context.
     #[tracing::instrument(level = "info", skip_all)]
     pub fn parse_view_item(&self, create_sql: String) -> Result<CatalogItem, anyhow::Error> {
-        let mut session_catalog = ConnCatalog {
-            state: Cow::Borrowed(self),
+        let mut state = self.clone();
+        // Existing catalog items might have been created while a
+        // specific feature flag turned on, so we need to ensure that this is also the
+        // case during catalog rehydration in order to avoid panics.
+        state
```

Comment:
There should be a function that sets this so it gets updated correctly when it changes. Also, looking at users of `for_system_session`, I believe it's safe (and maybe even correct) to do the WMR logic in that function for all callers. That could be useful for things like our AST migrations in the future that might similarly depend on these kinds of vars always being on if they once were on.

That's maybe a scarier change, so I'd be ok for now with a new function that makes a conncatalog from for_system_session and then sets WMR on it that the two parse fns call.